v3/src/components/data-display/components/legend/numeric-legend.tsx

```diff
@@ -44,7 +45,23 @@ export const NumericLegend =
       }
 
       if (choroplethElt) {
-        valuesRef.current = dataConfiguration?.numericValuesForAttrRole('legend') ?? []
+        /**
+         *  Adjust the value range displayed by the legend based on the data configuration model's properties:
+         *  1. If `displayOnlySelectedCases` is true and not all cases are visible, the legend displays the range of all
+         *     cases, both hidden and visible.
+         *  2. If `displayOnlySelectedCases` is false, and some cases are explicitly hidden by the user, the legend
+         *     displays the range of only the visible cases.
+         *  3. If all cases are hidden, the legend displays no range.
+         */
+        const allCasesCount = dataConfiguration?.dataset?.cases.length ?? 0
+        const hiddenCasesCount = dataConfiguration?.hiddenCases.length ?? 0
+        const allCasesHidden = hiddenCasesCount === allCasesCount
+        let values = allCasesHidden ? [] : dataConfiguration?.numericValuesForAttrRole("legend") ?? []
+        if (dataConfiguration?.displayOnlySelectedCases && !allCasesHidden && values.length < allCasesCount) {
+          const attribute = dataConfiguration?.dataset?.attrFromID(dataConfiguration?.attributeID("legend"))
+          values = attribute?.numValues ?? []
+        }
+        valuesRef.current = values
```

Comment:
I was having a pretty tough time understanding the logic here, even with your comments above. I wonder if something like this would be more clear? I also think you should put the cases in the comment in the same order they appear in the code, and you could even move each comment case to just above the conditional block to which it's referring.
```suggestion
        if (allCasesHidden) {
          valuesRef.current = []
        } else if (dataConfiguration?.displayOnlySelectedCases && values.length < allCasesCount) {
          const attribute = dataConfiguration?.dataset?.attrFromID(dataConfiguration?.attributeID("legend"))
          valuesRef.current = attribute?.numValues ?? []
        } else {
          valuesRef.current = dataConfiguration?.numericValuesForAttrRole("legend") ?? []
        }
```

Reply:
@tealefristoe Good points, thanks. I've updated things following your suggestions with just some very minor tweaks. I brought back a todo item to the comment block. I removed it because I thought it would add confusion, but I think it's probably best to leave it since it'll make it more likely the information in the comment gets updated when that part is added. 

Do you think that all looks good?